### PR TITLE
ENH: added `random.negative_binomial`

### PR DESCRIPTION
**Pull request by samir-nasibli**
## Description
negative_binomial(n, p[, size]) Draw samples from a negative binomial distribution.

## TODO:
* Adding supported devices into `Note`: is it needed??

## Reproduce
```python
>>> n, p = 10, .5  # n and p params
>>> np.mean(dpnp.random.negative_binomial(n, p, size=10**6))
10.006516
>>> np.var(dpnp.random.negative_binomial(n, p, size=10**6))
19.97882316627897
>>> np.mean(np.random.negative_binomial(n, p, size=10**6))
10.001372
>>> np.var(np.random.negative_binomial(n, p, size=10**6))
20.001929626350996
```

### CODE REVIEW

**Review by shssf — APPROVED** [review]

**Comment by oleksandr-pavlyk** [comment]
Please fix merge conflicts to enable reviewing.

File: dpnp/random/_random.pyx [hunk 1]
```diff
@@ -111,6 +113,35 @@ cpdef dparray dpnp_exponential(double beta, size):
...
+    # get the FPTR data structure
+    cdef DPNPFuncData kernel_data = get_dpnp_function_ptr(DPNP_FN_NEGATIVE_BINOMIAL, param1_type, param1_type)
+
+    result_type = dpnp_DPNPFuncType_to_dtype( < size_t > kernel_data.return_type)
+    # ceate result array with type given by FPTR data
+    cdef dparray result = dparray(size, dtype=result_type)
+
+    cdef fptr_custom_rng_negative_binomial_c_1out_t func = <fptr_custom_rng_negative_binomial_c_1out_t > kernel_data.ptr
+    # call FPTR function
+    func(result.get_data(), a, p, result.size)
```
Comment (oleksandr-pavlyk) [inline dpnp/random/_random.pyx:140]: Have you checked distribution parameters?

**Comment by samir-nasibli** [comment]
Will be further corrected according to the #236

File: dpnp/random/_random.pyx (hunk 1 above)
Comment (samir-nasibli) [reply dpnp/random/_random.pyx]: Added check and covered by tests

**Comment by samir-nasibli** [comment]
@oleksandr-pavlyk could you please review

**Review by shssf — APPROVED** [review]

**Pull request merged**